### PR TITLE
Deploy b network

### LINKED ISSUE #25: Deploy Upgradeable Smart Contract in Besu Network
Deploy Upgradeable Smart Contract in Besu Network.

Modify `trufle-config.json` if needed.

### PR DESCRIPTION
**Pull request by alejandroalffer**
Please, wait for merge

I'll try a fresh version of truflle, using HDWalletProvider and BIP39 mnemonic for Besu and GoQuroum based in Geth1.9 versions

### CODE REVIEW

**Comment by alejandroalffer** [comment]
Failed step 4. Looking for workarround

```
➜  temp-alastriaID-truffle-contracts git:(deploy-b-network) ✗ truffle compile

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.

➜  temp-alastriaID-truffle-contracts git:(deploy-b-network) ✗ truffle migrate --reset --compile-all --network "alastria-id-b"

Compiling your contracts...
===========================
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/identityManager/AlastriaIdentityIssuer.sol
> Compiling ./contracts/identityManager/AlastriaIdentityManager.sol
> Compiling ./contracts/identityManager/AlastriaIdentityServiceProvider.sol
> Compiling ./contracts/identityManager/AlastriaProxy.sol
> Compiling ./contracts/libs/Eidas.sol
> Compiling ./contracts/libs/Owned.sol
> Compiling ./contracts/nameService/AlastriaNameService.sol
> Compiling ./contracts/openzeppelin/GSN/Context.sol
> Compiling ./contracts/openzeppelin/Initializable.sol
> Compiling ./contracts/openzeppelin/ownership/Ownable.sol
> Compiling ./contracts/openzeppelin/upgradeability/AdminUpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/BaseAdminUpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/BaseUpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/Proxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/UpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/utils/Address.sol
> Compiling ./contracts/registry/AlastriaCredentialRegistry.sol
> Compiling ./contracts/registry/AlastriaPresentationRegistry.sol
> Compiling ./contracts/registry/AlastriaPublicKeyRegistry.sol
> Artifacts written to /Users/alejandro.alfonso/dev.alastria/temp-alastriaID-truffle-contracts/build/contracts
> Compiled successfully using:
   - solc: 0.5.17+commit.d19bba13.Emscripten.clang



Starting migrations...
======================
> Network name:    'alastria-id-b'
> Network id:      2020
> Block gas limit: 10485760 (0xa00000)


1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0x08e2796bd5875560e664650f47bd07f7b09456a599501f6f01a5b36f008fe1c7
   > Blocks: 0            Seconds: 4
   > contract address:    0xC4f9dfb165EeDd1958fbD2698a62d551893e37eb
   > block number:        15856450
   > block timestamp:     1622494357
   > account:             0x3cAa547890b7372f1fB4132753204d3ab4b2832D
   > balance:             0
   > gas used:            194533 (0x2f7e5)
   > gas price:           0 gwei
   > value sent:          0 ETH
   > total cost:          0 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:                   0 ETH


2_deploy_libraries.js
=====================

   Deploying 'Eidas'
   -----------------
   > transaction hash:    0x9ba4781ff4638974c3a62e258327b626b1ec500d80b0f525e3abcab7e9d0ff27
   > Blocks: 0            Seconds: 0
   > contract address:    0x97608f76809ec871Bb28E54971F9c2f6A3287CD1
   > block number:        15856452
   > block timestamp:     1622494361
   > account:             0x3cAa547890b7372f1fB4132753204d3ab4b2832D
   > balance:             0
   > gas used:            127657 (0x1f2a9)
   > gas price:           0 gwei
   > value sent:          0 ETH
   > total cost:          0 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:                   0 ETH


3_deploy_aid_contracts.js
=========================
Eidas address info saved!

   Linking
   -------
   * Contract: AlastriaIdentityIssuer <--> Library: Eidas (at address: 0x97608f76809ec871Bb28E54971F9c2f6A3287CD1)

   Linking
   -------
   * Contract: AlastriaIdentityManager <--> Library: Eidas (at address: 0x97608f76809ec871Bb28E54971F9c2f6A3287CD1)
serviceProvider deployed:  0xC59F0a9F2625Dd0e00c6762813F7b9F03465C162
AlastriaServiceProvider address info saved!
identityIssuer deployed:  0x7E994033089F9E9dbe80FA6A4Ce15798c75F3a1a
AlastriaIdentityIssuer address info saved!
credentialRegistry deployed:  0x9CbCf3BE4cd8EF90bB4753A2BDe3F519A123ef14
AlastriaCredentialRegistry address info saved!
presentationRegistry deployed:  0xAcfaeE7f8A06972b6472bA332BE4c6Db71eA83b8
AlastriaPresentationRegistry address info saved!
publicKeyRegistry deployed:  0x1A73Ba67232D24be8D44ec9c6B52CCAa413b9B69
AlastriaPublicKeyRegistry address info saved!
identityManager deployed:  0xEb843bb730692D011F51bD0acb9c50515C021847
AlastriaIdentityManager address info saved!

   > Saving migration to chain.
   -------------------------------------
   > Total cost:                   0 ETH


4_initialize_contracts.js
=========================
Credential registry initilized in 0x817f70d07a15fe81c639dff5c0a18a750169991ce959c39e7dcc931d50925cb7
Presentation registry initilized in 0xff321148b33c70454517350f994400c5102af451de918954c7411787798c7ecc
Public key registry initilized in 0x247a20c3a9ed463835cda9d8ef817b90533daa9f9c389a16b22bd3b6558df30c
Identity manager initilized in 0x1623c0066acbb845e9d55155fc57c06c0bcad77319395f186127d97bf4ab220c
Contracts initialized

TypeError: callback is not a function
    at module.exports (/Users/alejandro.alfonso/dev.alastria/temp-alastriaID-truffle-contracts/migrations/4_initialize_contracts.js:69:3)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at Migration._deploy (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:79:1)
    at Migration._load (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:61:1)
    at Migration.run (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:212:1)
    at Object.runMigrations (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:150:1)
    at Object.runFrom (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:110:1)
    at Object.runAll (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:114:1)
    at Object.run (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:79:1)
Truffle v5.3.7 (core: 5.3.7)
Node v14.16.1
➜  temp-alastriaID-truffle-contracts git:(deploy-b-network) ✗ 
```

**Comment by alejandroalffer** [comment]
Still failing at 6_xxx.json stage:

```
➜  temp-alastriaID-truffle-contracts git:(deploy-b-network) ✗ truffle migrate --reset --compile-all --network "alastria-id-b"

Compiling your contracts...
===========================
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/identityManager/AlastriaIdentityIssuer.sol
> Compiling ./contracts/identityManager/AlastriaIdentityManager.sol
> Compiling ./contracts/identityManager/AlastriaIdentityServiceProvider.sol
> Compiling ./contracts/identityManager/AlastriaProxy.sol
> Compiling ./contracts/libs/Eidas.sol
> Compiling ./contracts/libs/Owned.sol
> Compiling ./contracts/nameService/AlastriaNameService.sol
> Compiling ./contracts/openzeppelin/GSN/Context.sol
> Compiling ./contracts/openzeppelin/Initializable.sol
> Compiling ./contracts/openzeppelin/ownership/Ownable.sol
> Compiling ./contracts/openzeppelin/upgradeability/AdminUpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/BaseAdminUpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/BaseUpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/Proxy.sol
> Compiling ./contracts/openzeppelin/upgradeability/UpgradeabilityProxy.sol
> Compiling ./contracts/openzeppelin/utils/Address.sol
> Compiling ./contracts/registry/AlastriaCredentialRegistry.sol
> Compiling ./contracts/registry/AlastriaPresentationRegistry.sol
> Compiling ./contracts/registry/AlastriaPublicKeyRegistry.sol
> Artifacts written to /Users/alejandro.alfonso/dev.alastria/temp-alastriaID-truffle-contracts/build/contracts
> Compiled successfully using:
   - solc: 0.5.17+commit.d19bba13.Emscripten.clang



Starting migrations...
======================
> Network name:    'alastria-id-b'
> Network id:      2020
> Block gas limit: 10485760 (0xa00000)


1_initial_migration.js
======================

   Replacing 'Migrations'
   ----------------------
   > transaction hash:    0xc1af743cce74b72304aa78d237afa2721d57763e6a2495fcbb191d384bc245b6
   > Blocks: 0            Seconds: 4
   > contract address:    0x588e04ebF74584E97F6AA57Ea9F7E26c9CED4b91
   > block number:        15856864
   > block timestamp:     1622495737
   > account:             0x3cAa547890b7372f1fB4132753204d3ab4b2832D
   > balance:             0
   > gas used:            194533 (0x2f7e5)
   > gas price:           0 gwei
   > value sent:          0 ETH
   > total cost:          0 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:                   0 ETH


2_deploy_libraries.js
=====================

   Replacing 'Eidas'
   -----------------
   > transaction hash:    0xee8c5edd928bb752f43803e5cfd143d566ea6cb1ca58445b4654aa695c194863
   > Blocks: 0            Seconds: 0
   > contract address:    0x2A83dAA25FD6609bdfabd509740ED21B620E9Ed9
   > block number:        15856867
   > block timestamp:     1622495743
   > account:             0x3cAa547890b7372f1fB4132753204d3ab4b2832D
   > balance:             0
   > gas used:            127657 (0x1f2a9)
   > gas price:           0 gwei
   > value sent:          0 ETH
   > total cost:          0 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:                   0 ETH


3_deploy_aid_contracts.js
=========================
Eidas address info saved!

   Linking
   -------
   * Contract: AlastriaIdentityIssuer <--> Library: Eidas (at address: 0x2A83dAA25FD6609bdfabd509740ED21B620E9Ed9)

   Linking
   -------
   * Contract: AlastriaIdentityManager <--> Library: Eidas (at address: 0x2A83dAA25FD6609bdfabd509740ED21B620E9Ed9)
serviceProvider deployed:  0xba86EA3585d697d38cC0364689a75bccdb492f5B
AlastriaServiceProvider address info saved!
identityIssuer deployed:  0x189A9E8e12D77b2bb551779834a6d5B801c7832b
AlastriaIdentityIssuer address info saved!
credentialRegistry deployed:  0x7541C5746FD0c50744Bdf23565bF8B997F82e52e
AlastriaCredentialRegistry address info saved!
presentationRegistry deployed:  0x655f10CDAB817872B322Bc95674EcfD277d30738
AlastriaPresentationRegistry address info saved!
publicKeyRegistry deployed:  0x46Cd9F66bD72Dd6c6482727234F2e9A610a1d62f
AlastriaPublicKeyRegistry address info saved!
identityManager deployed:  0xa7eF7a8E6AaC3fb64F71964FfCc5C3Bd48ebF88E
AlastriaIdentityManager address info saved!

   > Saving migration to chain.
   -------------------------------------
   > Total cost:                   0 ETH


4_initialize_contracts.js
=========================
Credential registry initilized in 0xd845d688ca0aba696ef13349f17e16bdc115f75ff2ec9cbd6f8fc1bb94e519d2
Presentation registry initilized in 0x287adfe44b3ab985a7c7695126d92521057cee21737c01ae8c78849df224f530
Public key registry initilized in 0x0258a9b02ea171afdb1cd73121c534725b4dc382a0aa9d9a41ef72ad85ba938e
Identity manager initilized in 0xcc2234b2bc04da15387716af86201e8957b861b0219dd317916c153af7cb2bc5
Contracts initialized

   > Saving migration to chain.
   -------------------------------------
   > Total cost:                   0 ETH


5_deploy_ans_contracts.js
=========================
proxy first identity wallet: 0xCF1d0395fadd0EBdCeFC9f3fd434E53502dE8Ba2

   Deploying 'AlastriaNameService'
   -------------------------------
   > transaction hash:    0x9aa9bf318836ec515554276d15d6f749ff16e490a5332809dcb92acb6c41aa3e
   > Blocks: 0            Seconds: 0
   > contract address:    0x4e783b002117f3A982C3E794A84A06BC9c03d6bF
   > block number:        15856896
   > block timestamp:     1622495841
   > account:             0x3cAa547890b7372f1fB4132753204d3ab4b2832D
   > balance:             0
   > gas used:            1851952 (0x1c4230)
   > gas price:           0 gwei
   > value sent:          0 ETH
   > total cost:          0 ETH

nameService deployed:  0x4e783b002117f3A982C3E794A84A06BC9c03d6bF
AlastriaNameService address info saved!

   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:                   0 ETH


6_update_logic.js
=================

Transaction: 0x5abf009ae726216e19eedfd5fd4b4e18d8f74750c6a64aba256f2efe90c9275b exited with an error (status 0). 
     Please check that the transaction:
     - satisfies all conditions set by Solidity `require` statements.
     - does not trigger a Solidity `revert` statement.

Truffle v5.3.7 (core: 5.3.7)
Node v14.16.1
```

File: package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
   "license": "MIT",
   "dependencies": {
     "@truffle/config": "1.2.33",
-    "@truffle/hdwallet-provider": "1.1.1",
+    "@truffle/hdwallet-provider": "^1.1.1",
```
Comment (nefera606) [inline package.json:36]: We need to be so careful with the ^ versions, we need to bring support to all the future changes on it.

**Comment by VictorNS69** [comment]
This PR will continue in #33

**Pull request closed without merging**